juju/juju
# Remove Method: SetDevicesAddressesIdempotently

## Description of change

The method `SetDevicesAddressesIdempotently` had some convoluted logic around ensuring the integrity of provider IDs for link-layer device addresses.

Recent changes mean that this method is no longer supplied with providers IDs, so we can simply remove it and use `SetDevicesAddresses` instead.

Included are removal of methods unused as a result, plus some minor spelling/casing fixes.

## QA steps

- Bootstrap to AWS.
- `juju add-machine`.
- `juju add-machine lxd:0`.
- Check that the logs have no associated errors.

## Documentation changes

None.

## Bug reference

N/A
